project_quazar/lib/project_quazar/game/ship.ex

```diff
@@ -132,11 +132,21 @@ defmodule Ship do
 
   @doc """
   Fires a bullet from the ship, creating a new `Bullet` instance.
-  Returns a `Bullet` struct representing the new bullet, or an error if the bullet type is invalid.
+  The bullet is accelerated by its predefined speed.
+  Returns `{:ok, bullet}` where `bullet` is the newly created `Bullet` struct, or `{:error, reason}` if the bullet type is invalid.
   """
-  def fire(%__MODULE__{kinematics: kinematics, bullet_type: bullet_type}, player_name) do
+  def fire(%__MODULE__{kinematics: kinematics, radius: radius, bullet_type: bullet_type}, player_name) do
     %{px: px, py: py, vx: vx, vy: vy, angle: angle} = kinematics
-
-    Bullet.new_bullet(player_name, px, py, vx, vy, angle, bullet_type)
+    # Bullet will start at the edge of the ship, in the direction the ship is facing.
+    ship_front_x = px + radius
+    ship_front_y = py + radius
```

Comment:
We will fix this later, should use the current ship angle to determine where to offset the bullet.